From 31dd5ff81b6bb06a3c02875042898e70525baf38 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 31 May 2022 09:04:53 +0300
Subject: [PATCH 23/30] configs: s32cc: Pass a copy of DTB to distro build

Make a copy of the U-Boot dtb and pass it to the next EFI component
instead of operating on the original DTB. This is needed to avoid
failures caused by the content alteration during distro boot.
E.g. Removing properties from the mmc node causes U-Boot failures
while loading files.

Issue: ALB-8903
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32-cc.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/include/configs/s32-cc.h b/include/configs/s32-cc.h
index b8bdffaf20..23bdb66430 100644
--- a/include/configs/s32-cc.h
+++ b/include/configs/s32-cc.h
@@ -315,7 +315,10 @@
  * Variables required by doc/README.distro
  */
 #  define DISTRO_VARS \
-	"setenv fdt_addr ${fdtcontroladdr};" \
+	"fdt addr ${fdtcontroladdr};" \
+	"fdt header get fdt_size totalsize;" \
+	"cp.b ${fdtcontroladdr} ${fdt_addr} ${fdt_size};" \
+	"setenv fdt_addr_r ${fdt_addr};" \
 	"setenv ramdisk_addr_r " __stringify(S32CC_RAMDISK_ADDR) ";" \
 	"setenv kernel_addr_r ${loadaddr};" \
 	"setenv pxefile_addr_r " __stringify(S32CC_PXE_ADDR) ";" \
@@ -324,7 +327,7 @@
  * Remove pinmuxing properties as SIUL2 driver isn't upstreamed yet
  */
 #  define DISTRO_FIXUPS \
-	"fdt addr ${fdtcontroladdr};" \
+	"fdt addr ${fdt_addr_r};" \
 	"fdt rm serial0 pinctrl-0;" \
 	"fdt rm serial0 pinctrl-names;" \
 	"fdt rm mmc0 pinctrl-0;" \
-- 
2.17.1

